src/util/CartesianUtils.ts

```diff
@@ -32,15 +32,43 @@ export const formatAxisMap = (props: any, axisMap: any, offset: any, axisType: A
     const { orientation, domain, padding = {}, mirror, reversed } = axis;
     const offsetKey = `${orientation}${mirror ? 'Mirror' : ''}`;
 
-    let range, x, y, needSpace;
+    let calculatedPadding, range, x, y, needSpace;
+
+    if (axis.padding === 'gap' || axis.padding === 'no-gap') {
+      const diff = domain[1] - domain[0];
+      let smallestDistanceBetweenValues = Infinity;
+      const sortedValues = axis.categoricalDomain.sort();
```

Comment:
There is an assumption here, the type of `axis` must be `number` when we `axis.padding === 'gap' || axis.padding === 'no-gap'`?

Reply:
fixed